router/batchrouter/asyncdestinationmanager/salesforce-bulk/api_service.go

```diff
@@ -0,0 +1,279 @@
+package salesforcebulk
+
+import (
+	"bytes"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/rudderlabs/rudder-go-kit/jsonrs"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	oauthv2 "github.com/rudderlabs/rudder-server/services/oauth/v2"
+	cntx "github.com/rudderlabs/rudder-server/services/oauth/v2/context"
+)
+
+func NewSalesforceAPIService(
+	logger logger.Logger,
+	destinationInfo *oauthv2.DestinationInfo,
+	apiVersion string,
+	client *http.Client,
+) *SalesforceAPIService {
+	return &SalesforceAPIService{
+		logger:          logger,
+		destinationInfo: destinationInfo,
+		apiVersion:      apiVersion,
+		client:          client,
+	}
+}
+
+func (s *SalesforceAPIService) CreateJob(
+	objectName, operation, externalIDField string,
+) (string, *APIError) {
+	reqBody := JobCreateRequest{
+		Object:      objectName,
+		ContentType: "CSV",
+		Operation:   operation,
+		LineEnding:  "LF",
+	}
+
+	if operation == "upsert" && externalIDField != "" {
+		reqBody.ExternalIDFieldName = externalIDField
+	}
+
+	body, err := jsonrs.Marshal(reqBody)
+	if err != nil {
+		return "", &APIError{
+			StatusCode: 500,
+			Message:    fmt.Sprintf("marshalling job request: %v", err),
+			Category:   "ServerError",
+		}
+	}
+
+	endpoint := fmt.Sprintf("https://default.salesforce.com/services/data/%s/jobs/ingest", s.apiVersion)
```

Comment:
Why default?

Reply:
This is just a placeholder. We are updating it later through the augmenter.

Reply:
You mean this right?

```GO
	r.Header.Set("Authorization", fmt.Sprintf("Bearer %s", accessToken))
	r.URL.Host = instanceURL
	r.Body = io.NopCloser(bytes.NewReader(body))
```